Javatrix.java

```diff
@@ -477,5 +477,37 @@ public Javatrix times(double scalar)
         }
         return actual;
     }
+
+    /**
+     * Get a submatrix.
+     *
+     * @param r - an array of row indices
+     * @param c - an array of column indices
+     * @return submatrix(r(:), c(:))
+     * @throws ArrayIndexOutOfBoundsException if submatrix
+     * has larger dimensions than this.matrix.
+     */
+    public Javatrix  getMatrix(int[] r, int[] c)
+        throws ArrayIndexOutOfBoundsException
+    {
+        m = r.length;
+        n = c.length;
+        int row;
+        int col;
+        if (m > this.getRowDimension() || n > this.getColumnDimension())
```

Comment:
Since the row location in the submatrix is actually a value from the 1D array r, and the column location in the submatrix is a value from c, I think the test needs to check that any of the **values** in each of the arrays are within the size of the calling Javatrix object (m and n) - if they are not, then the exception should be thrown. The length of each of the arrays does not come into play as far as I can tell? 

I would also move the exception test to the top - it's more efficient to check if the exception is going to be thrown before declaring and initializing variables that won't be used if it is thrown.